packages/waku/tests/tsconfig.json

```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "../",
+    "noEmit": true
+  },
+  "include": ["../src", "."]
```

Comment:
`src` is already checked, so this is duplicated, right?
Maybe we should move this file to `packages/waku/tsconfig.test.json`. What do you think?

Reply:
as `tests/tsconfig.json` vscode (and any other editor with a typescript server) will find the tsconfig for the tests and use it. (otherwise it reads from `packages/waku/tsconfig.json` which excludes the tests, so type checking does not work. This is why we saw bad imports and no enforcement of the `import type {` specifier before. vscode would need custom configuration to acknowledge `tsconfig.test.json` and so would every other editor. 

`"include": ["../src"]`  is needed so we can import from `../src` which typescript considers a separate "project" unless we include it in the same tsconfig.

![image](https://github.com/user-attachments/assets/97af1648-8a96-451d-8e15-d4924c8c21d9)

Reply:
alright.